dump-event-schema-compiler-plugin/src/main/scala/org/virtuslab/akkasaferserializer/DumpEventSchemaPluginComponent.scala

```diff
@@ -0,0 +1,97 @@
+package org.virtuslab.akkasaferserializer
+import org.virtuslab.akkasaferserializer.writer.SchemaWriter
+import org.virtuslab.akkasaferserializer.model.{Field, TypeDefinition}
+
+import scala.tools.nsc.{Global, Phase}
+import scala.tools.nsc.plugins.PluginComponent
+
+class DumpEventSchemaPluginComponent(val options: DumpEventSchemaOptions, val global: Global) extends PluginComponent {
+  import global._
+  override val phaseName: String = "dump-event-schema"
+  override val runsAfter: List[String] = List("typer")
+
+  lazy val writer = new SchemaWriter(options)
+
+  override def newPhase(prev: Phase): Phase =
+    new StdPhase(prev) {
+
+      private val effectNames =
```

Comment:
In the longer run (https://github.com/VirtusLab/akka-safer-serializer/issues/36) more classes will be included here, not just effects... so I'd call it `genericsNames` for consistency with the other plugin